peachjam/adapters/adapters.py

```diff
@@ -76,26 +76,52 @@ def __init__(self, settings):
             }
         )
         self.url = self.settings["url"]
+        self.api_url = self.settings["api_url"]
+        self.locality = self.settings.get(
+            "locality"
+        )  # using .get to avoid KeyError since locality is not required
+        self.jurisdiction = self.settings["jurisdiction"]
+        self.place_code = f"{self.jurisdiction}{'-' if self.locality else ''}{self.locality if self.locality else ''}"
 
     def check_for_updates(self, last_refreshed):
         """Checks for documents updated since last_refreshed (which may be None), and returns a list
         of document identifiers (expression FRBR URIs) which must be updated.
         """
-        return self.get_updated_documents(last_refreshed)
+        docs = self.get_doc_list()
+        updated_docs = self.check_for_updated(docs, last_refreshed)
+        deleted_docs = self.check_for_deleted(docs)
+
+        return updated_docs, deleted_docs
 
     def get_doc_list(self):
-        url = self.url
+        url = f"{self.api_url}/akn/{self.place_code}/.json"
         results = []
         while url:
             res = self.client_get(url).json()
             results.extend(res["results"])
             url = res["next"]
         return results
 
-    def get_updated_documents(self, last_refreshed):
+    def check_for_deleted(self, docs):
+        uris = {d["expression_frbr_uri"] for d in docs}
+        for doc in docs:
+            for pit in doc["points_in_time"]:
+                for expr in pit["expressions"]:
+                    uris.add(expr["expression_frbr_uri"])
+
+        qs = Legislation.objects.filter(jurisdiction__iso=self.jurisdiction.upper())
+        if self.locality:
+            qs = qs.filter(locality__code=self.locality)
```

Comment:
if there is no locality, you must explicitly filter on `locality=None`.

Reply:
Added. :heavy_check_mark:

Reply:
did you git push?

Reply:
sorry, check now